### ISSUE TITLE
Color of interaction hint makes it difficult to see

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/quadrilateral#348]
I noticed the teal interaction hint on the blue background was a little hard to see. Here is a colorblindness simulator (no idea if it is legit...)
![image](https://user-images.githubusercontent.com/6396244/216701709-1ce876ee-1c30-414b-807e-269e7a238885.png)

@BLFiedler is this an issue? Feel free to close if not.

**Comment by BLFiedler** [comment]
Agreed. Green-blind is a little rough too. Let me see if there's anything that works a bit better across all.

![image](https://user-images.githubusercontent.com/47331395/217034249-6205f28f-246d-4af2-b6b7-c95542b12bb1.png)

**Comment by BLFiedler** [comment]
To mesh with the background, we can't really go more blue/purple, and more yellow is mostly too much like the shape color. Green will have to be the one. If we go for more of a "mint green" by desaturating, it is fairly contrasting across monochrome, green blind (and red blind as well).

Value used: #b5ffd8, rgb(181, 255, 216)

![image](https://user-images.githubusercontent.com/47331395/217053878-eb6fdad8-0633-473e-8a5b-5c4386ab7bb9.png)

![image](https://user-images.githubusercontent.com/47331395/217053927-06703c89-67da-47cc-88e2-aa5b9ec2fc5e.png)

![image](https://user-images.githubusercontent.com/47331395/217054013-807c6319-ff3c-48d3-8840-488b298cd215.png)


Whaddya think? @jessegreenberg

**Comment by BLFiedler** [comment]
Made the change above. I'll keep this open a bit longer assigned to me in case I have a moment to ponder a better interaction cue.

**Comment by terracoda** [comment]
Sharing some WCAG resources that I am reviewing to see if they are helpful.
https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html

**Comment by terracoda** [comment]
I would be more concerned about how good the contrast is between the focus highlight (and Interactive Highlights) is when compared to the background color.

<img width="439" alt="image" src="https://user-images.githubusercontent.com/1902116/222193979-c6070e4d-aba7-4f95-8674-e07e6ac71b80.png">

Quoting guide:
> The intent of this Success Criterion is to ensure that active user interface components (i.e., controls) and meaningful graphics are distinguishable by people with moderately low vision. 

It is important to consider that in our sims in addition to good contrast, people can zoom in and people can use focus highlights (or interactive highlights). In addition, there is Voicing that can be enabled to assist a learner with additional needs.

> This success criterion does not require that controls have a visual boundary indicating the hit area, but if the visual indicator of the control is the only way to identify the control, then that indicator must have sufficient contrast. 

The dashed visual hint is not the only indicator that the corners an sides are interactive. I think we would document that actual contrast ratios and make decisions from there.

Regarding conflict between visual hint and the grid, we could consider other colors if the grid is enabled, but honestly the shape itself has black outlines, and the pink highlights would still have high contrast over the grid, so I kind of feel like we are good.

Can we note the actual contrast between the current colors?
- background: visual hint
- background: pink highlight

**Comment by terracoda** [comment]
Ok, the foreground-background contrast ratio is low. We should be aiming for 3:1 for user interface components.
<img width="685" alt="image" src="https://user-images.githubusercontent.com/1902116/222209906-b42bb6b4-4f0b-4463-bbc5-a1b0005a2451.png">

Is the grid white? We obviously need a dark-ish background for the grid.
I don't want us to panic on this. I need to understand what colors are essential?

**Comment by BLFiedler** [comment]
If you'd like to, you can directly play with the colors by changing this file: quadrilateral\js\QuadrilateralColors.ts

**Comment by BLFiedler** [comment]
Could go this direction.

![image](https://user-images.githubusercontent.com/47331395/222238166-35901fb8-3cc6-44d1-9e21-546875e2873e.png)
![image](https://user-images.githubusercontent.com/47331395/222238265-06bef509-920c-4627-a415-8702f8550a58.png)
![image](https://user-images.githubusercontent.com/47331395/222238327-ac6f84a4-0ed0-4d94-9560-acd2efce9ff8.png)

worst contrast comes in blue-blind:
![image](https://user-images.githubusercontent.com/47331395/222238106-721c6616-4717-4405-80a7-e0e10dc00a7a.png)


monochrome isn't terrible:
![image](https://user-images.githubusercontent.com/47331395/222238037-7984fe18-31f9-44ba-ba51-617f29379a75.png)


```
  screenBackgroundColorProperty: new ProfileColorProperty( quadrilateral, 'background', {
    default: new Color( '#C4A484' )
  } ),

  quadrilateralShapeColorProperty: new ProfileColorProperty( quadrilateral, 'quadrilateralShape', {
    default: new Color( '#FFD700' )
  } ),

  interactionCueColorProperty: new ProfileColorProperty( quadrilateral, 'interactionCueColor', {
    default: new Color( '#7DF9FF' )
  } )
```

**Comment by terracoda** [comment]
Further reading... the outline color of the shape is black, so it is in high contrast between both the shape color and the background color. 

The shape is visible to people with low vision. We are exploring a background color that might provide a better contrast with the pink highlights and visual interactive hints.

We also thickened the diagonal guides so they are easier to see when aligned with a gridline.

**Comment by BLFiedler** [comment]
Changes made locally. Will push separate commits for each change later after I get latest changes into my copy.

**Comment by terracoda** [comment]
From the top of the success criteria documentation
> The visual [presentation](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html#dfn-presentation) of the following have a [contrast ratio](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html#dfn-contrast-ratio) of at least 3:1 against adjacent color(s):
> Graphical Objects
    Parts of graphics required to understand the content, except when a particular presentation of graphics is [essential](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html#dfn-essential) to the information being conveyed.

The outline of the shape is in high contrast with the adjacent colors (background color and shape color), so we are totally good there.

**Comment by BLFiedler** [comment]
Hadn't realized I had these commits in the push I made (had some git VSCode integration issues...) , but worth reviewing anyway. 

This is a big color change, but does represent some review @terracoda and I did together a couple of weeks ago. Contrast is good now with this slightly less bright color. We made some thickness changes to various borders/grid lines/diagonals/interaction cues as well to help with contrast. We tried not to change the shape color, but 

![image](https://user-images.githubusercontent.com/47331395/225697391-f5d07276-d2d2-4da9-a7d6-3fafd5826794.png)

![image](https://user-images.githubusercontent.com/47331395/225697447-ecf4a08e-cb07-4967-9b2b-49f18ace8e1d.png)

![image](https://user-images.githubusercontent.com/47331395/225698102-0fdf06fb-ab1d-44fb-9fae-f892c06ca5a1.png)

![image](https://user-images.githubusercontent.com/47331395/225698145-1b5374fa-a176-4d56-a74b-8a80a52a3382.png)


Calling all interested parties to see how they feel about it: @terracoda @Ashton-Morris @emily-phet

**Comment by BLFiedler** [comment]
Oh, also I now realize I used hex codes for the colors for copy-paste expediency from the color picker. Is that alright, @jessegreenberg, or should I go back and convert to RGB for consistency?

**Comment by jessegreenberg** [comment]
Should be totally fine to use hex codes.

**Comment by jessegreenberg** [comment]
Those filter tests look really nice! But I have to admit I am not crazy about the yellow shape + tan background combo.

**Comment by BLFiedler** [comment]
Going the other direction! White is where we should have started. We compared using a white background with using a light blue like Calculus Grapher. Both are fine.

It might be a good time to try out having the play area and the rest of the screen background be different colors. 

@jessegreenberg indicated this is a simple change to make the grid fill be adjustable in quadrilateral\js\QuadrilateralColors.ts.  
Go for it!

We'll try out the white fill and light blue (#F3FCFE) for the control area. I have a multitude of color changes saved locally for borders and grid lines to accommodate the swap from dark to light background, so I can take care of those changes once the property is accessible from QuadrilateralColors.ts.

**Comment by jessegreenberg** [comment]
OK, I added a `gridFillColorProperty` and set to white. Then I made the background fill color the suggested light blue. Just so things are still visible I also changed:

- gridLinesColorProperty to black
- diagonalGuidesStrokeColorProperty to black
- gridStrokeColorProperty to black
- interactionCueColorProperty to blue

Back to @BLFiedler and @terracoda to play with other colors now that grid and background are separated.

Here is what it looks like now:
![image](https://user-images.githubusercontent.com/6396244/226070387-5366bde1-eef6-4ad6-9052-2e4ecccd37e2.png)

I think it is nice and will match calculus-grapher! But I like that we have flexibility to make the background color more fun if we want since it doesn't have to match the grid fill.

**Comment by jessegreenberg** [comment]
Sorry, looks like I did not push this on Friday. Pushed.
